net/etesync-server/files/init.uwsgi

```diff
@@ -0,0 +1,158 @@
+#!/bin/sh /etc/rc.common
+
+START=80
+
+USE_PROCD=1
+
+INI="; this file is autmatically re-created from /etc/config/etesync_server\n"
+
+IFACES=""
+
+NGINX_CHANGED=0
+
+config_nginx_for_domain() { # assuming "$1" == "[0-9A-za-z.-]*"
+    local domain="$1"
+
+    local dom="${domain%.}.?"
+    while [ "$dom" != ".?" ]; do
+
+        local rgx="${dom//./[.]}"
+        rgx="^\s*(server_name|\"server_name\"|'server_name')(\s.*)?" \
+            "\s(${rgx}|\"${rgx}\"|'${rgx}')(\s.*)?\s*;"
+
+        local prefer=""
+        local conf=""
+
+        for conf in $(grep -lE "${rgx}" /etc/nginx/conf.d/*.conf); do
+            grep -qE '^\s*include\s+etesync.locations\s*;' "${conf}" && return
+            grep -qE '^\s*ssl_certificate\s' "${conf}" && prefer="${conf}"
+        done
+
+        [ -z "${prefer}" ] \
+        && echo "Consider adding a self-signed certifacate by: " \
+                "	nginx-util add_ssl '$(basename "${conf}" .conf)'" \
+        || conf="${prefer}"
+
+        [ -n "${conf}" ] && { # use prefered or last file containing ${rgx}.
+            echo "Adding 'include etesync.locations;' to: ${conf}"
+            sed -iE "/${rgx}/a	include etesync.locations;" "${conf}" \
+            && NGINX_CHANGED=1
+
+            return
+        }
+
+        [ "${dom:0:3}" == "[*]" ] && dom="${dom:3}" || dom="[*].${dom#?*.}"
+    done
+
+    # no conf file with suitable server_name:
+    local conf="/etc/nginx/conf.d/${domain}.conf"
+    echo "Creating ${conf} with 'include etesync.locations;' ..."
+    sed -E "s/DOMAIN;/'${domain}';/g" \
+        /usr/share/etesync-server/nginx.conf.template > "${conf}" \
+    && nginx-util add_ssl "${domain}" \
+    && NGINX_CHANGED=1
+}
+
+
+config_allowed_hosts_indirectly() {
+    option_cb() { return; }
+
+    list_cb() {
+        [ "$1" == "allow_all_ips_of" ] && {
```

Comment:
== is a bashism. Shellcheck can help here.

Reply:
Changed it.